src/v/rpc/connection_cache.h

```diff
@@ -116,6 +116,25 @@ class connection_cache final
           std::forward<Func>(f));
     }
 
+    /// If a reconnect_transport is in a backed-off state, reset
+    /// it so that the next RPC will be dispatched.  This is useful
+    /// when a down node comes back to life: the first time we see
+    /// a message from a re-awakened peer, we reset their backoff.
+    ss::future<> reset_client_backoff(
+      model::node_id self, ss::shard_id src_shard, model::node_id node_id) {
+        auto shard = rpc::connection_cache::shard_for(self, src_shard, node_id);
+
+        return container().invoke_on(
+          shard, [node_id](rpc::connection_cache& cache) mutable {
+              if (!cache.contains(node_id)) {
+                  // No client available
+                  return;
+              }
+              auto recon_transport = cache.get(node_id);
+              recon_transport->reset_backoff();
```

Comment:
It seems that `reset_backoff` isn't defined yet, you add it only in c336f5f. Can you please make each commit compile-able so we could use `git bisec` if we need to in the future.